storage/operation/stats.go

```diff
@@ -0,0 +1,168 @@
+package operation
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module/util"
+	"github.com/onflow/flow-go/storage"
+)
+
+// Stats holds statistics for a single prefix group.
+type Stats struct {
+	Count       int
+	TotalSize   int
+	MinSize     int
+	MaxSize     int
+	AverageSize float64
+}
+
+// SummarizeKeysByFirstByteConcurrent iterates over all prefixes [0x00..0xFF] in parallel
+// using nWorker goroutines. Each worker handles one prefix at a time until all are processed.
+//
+// The storage.Reader must be able to create multiple iterators concurrently.
+func SummarizeKeysByFirstByteConcurrent(log zerolog.Logger, r storage.Reader, nWorker int) (map[byte]Stats, error) {
+	// We'll have at most 256 possible prefixes (0x00..0xFF).
+	// Create tasks (one per prefix), a results channel, and a wait group.
+	taskChan := make(chan byte, 256)
+	resultChan := make(chan struct {
+		prefix byte
+		stats  Stats
+		err    error
+	}, 256)
+
+	var wg sync.WaitGroup
+
+	// Start nWorker goroutines.
+	for i := 0; i < nWorker; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for prefix := range taskChan {
+				st, err := processPrefix(r, prefix)
+				resultChan <- struct {
+					prefix byte
+					stats  Stats
+					err    error
+				}{
+					prefix: prefix,
+					stats:  st,
+					err:    err,
+				}
+			}
+		}()
+	}
+
+	progress := util.LogProgress(log,
+		util.DefaultLogProgressConfig(
+			"Summarizing keys by first byte",
+			256,
+		))
+
+	// Send all prefixes [0..255] to taskChan.
+	for p := 0; p < 256; p++ {
+		taskChan <- byte(p)
+	}
+	close(taskChan)
+
+	// Once all workers finish, close the result channel.
+	go func() {
+		wg.Wait()
+		close(resultChan)
+	}()
+
+	// Gather results. We'll accumulate them in a map[prefix]Stats.
+	finalStats := make(map[byte]Stats, 256)
+
+	// If we encounter an error, we will return it immediately.
+	for res := range resultChan {
+		if res.err != nil {
+			return nil, res.err
+		}
```

Comment:
We should probably shut down running goroutines that reads db before exiting this function on error.  We want to avoid `storage.Reader` being used in goroutines after `db.Close()` is called in main function.